mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -1,409 +1,74 @@
-from packaging.version import Version
-import logging
-import mlflow.pytorch
-import os
-import shutil
-import tempfile
-import pytorch_lightning as pl
-from pytorch_lightning.utilities import rank_zero_only
-
-from mlflow.exceptions import MlflowException
-from mlflow.utils.autologging_utils import (
-    ExceptionSafeAbstractClass,
-    BatchMetricsLogger,
-    MlflowAutologgingQueueingClient,
-    get_autologging_config,
+import time
+import mlflow
+from mlflow.tracking import MlflowClient
+from mlflow.entities import Param, Metric
+from mlflow.tensorflow import (
+    _flush_queue,
+    _metric_queue,
+    _metric_queue_lock,
+    _thread_pool,
+    _MAX_METRIC_QUEUE_SIZE,
 )
 
-logging.basicConfig(level=logging.ERROR)
-
-
-# The following are the downsides of using PyTorch Lightning's built-in MlflowLogger.
-# 1. MlflowLogger doesn't provide a mechanism to store an entire model into mlflow.
-#    Only model checkpoint is saved.
-# 2. For storing the model into mlflow `mlflow.pytorch` library is used
-# and the library expects `mlflow` object to be instantiated.
-# In case of MlflowLogger, Run management is completely controlled by the class and
-# hence mlflow object needs to be reinstantiated by setting
-# tracking uri, experiment_id and run_id which may lead to a race condition.
-# TODO: Replace __MlflowPLCallback with Pytorch Lightning's built-in MlflowLogger
-# once the above mentioned issues have been addressed
-
-
-_pl_version = Version(pl.__version__)
-if _pl_version < Version("1.5.0"):
-    # pylint: disable-next=ungrouped-imports
-    from pytorch_lightning.core.memory import ModelSummary
-else:
-    # pylint: disable-next=ungrouped-imports
-    from pytorch_lightning.utilities.model_summary import ModelSummary
-
-
-def _get_optimizer_name(optimizer):
-    """
-    In pytorch-lightining 1.1.0, `LightningOptimizer` was introduced:
-    https://github.com/PyTorchLightning/pytorch-lightning/pull/4658
-
-    If a user sets `enable_pl_optimizer` to True when instantiating a `Trainer` object,
-    each optimizer will be wrapped by `LightningOptimizer`:
-    https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.trainer.trainer.html
-    #pytorch_lightning.trainer.trainer.Trainer.params.enable_pl_optimizer
-    """
-    if Version(pl.__version__) < Version("1.1.0"):
-        return optimizer.__class__.__name__
-    else:
-        # pylint: disable-next=ungrouped-imports
-        from pytorch_lightning.core.optimizer import LightningOptimizer
 
-        return (
-            optimizer._optimizer.__class__.__name__
-            if isinstance(optimizer, LightningOptimizer)
-            else optimizer.__class__.__name__
-        )
+ENABLED = True  # set to False while inside lightning's patched_fit()
```

Comment:
Can we do something like this?

```suggestion
DISABLED = False

def disable_pytorch_autlogging():
    global DISABLED
    
    DISABLED = False
    
    # Call this in patched_fit
```

Reply:
i implemented this as a context manager now